Improve documentation regarding artifacthub.io annotations (#928)
adfinis/helm-charts: docs/development.md

pre-commit install
pre-commit install-hooks
```

### Changelog

> We're adding [Artifact Hub annotations](https://artifacthub.io/docs/topics/annotations/helm/) to our charts to improve the visibility of changes for our charts. These annotations are added to `Chart.yaml` as a YAML string and need to be replaced on every update, as they are not additive but specific to this release.

Reviewer: ```suggestion
We add [Artifact Hub annotations](https://artifacthub.io/docs/topics/annotations/helm/) to our charts to improve the visibility of changes. These annotations are added to `Chart.yaml` as a YAML string and need to be replaced on every update, as they are not additive but specific to this release.
```

Author: Yes, definitely better like that.

```yaml
annotations:
  artifacthub.io/changes: |
    - kind: changed

>       description: "feat(argo-rollouts): Upgrade to argo-rollouts 1.3.0"

Reviewer: is this a good example? it's lazy copypasta from upstreams git history but also highly redundant. maybe our example could be more concise:
```suggestion
      description: "feat(argo-rollouts): Update to 1.3.0"
```
or even just this:
```suggestion
      description: "argo-rollouts: Update to 1.3.0"
```

Author: What do you think about the examples now?

      links:
        - name: GitHub PR
          url: https://github.com/argoproj/argo-helm/pull/1466
    - kind: changed
      description: "fix(argo-rollouts): Upgrade argo-rollouts to 1.3.1"
      links:

>         - name: GitHub PR

Author: I think I'm going to extend the examples with cases for both.
